File: apps/mobile/lib/features/community/domain/models/vet_request_details.dart
import 'package:meta/meta.dart';

@immutable
class VetRequestDetails {
  final String reasonForVisit;
  final String? clinicName;
  final DateTime? appointmentDate;
  final bool transportNeeded;

  const VetRequestDetails({
    required this.reasonForVisit,
    this.clinicName,
    this.appointmentDate,
    required this.transportNeeded,
  });
}

File: apps/api/HappyPaws.Domain/Entities/VetRequestDetails.cs
using System;

namespace HappyPaws.Domain.Entities;

public class VetRequestDetails
{
    public Guid PostId { get; set; }
    public string ReasonForVisit { get; set; } = null!;
    public string? ClinicName { get; set; }
    public DateTimeOffset? AppointmentDate { get; set; }
    public bool TransportNeeded { get; set; } = false;

    public Post Post { get; set; } = null!;
}
